pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperator.java

```diff
@@ -67,19 +67,21 @@ public class MailboxReceiveOperator extends MultiStageOperator {
   private int _serverIdx;
   private TransferableBlock _upstreamErrorBlock;
 
-  private static MailboxIdentifier toMailboxId(VirtualServer sender, long jobId, long stageId,
-      VirtualServerAddress receiver) {
+  private static MailboxIdentifier toMailboxId(VirtualServer sender, long jobId, int senderStageId,
+      int receiverStageId, VirtualServerAddress receiver) {
     return new JsonMailboxIdentifier(
-        String.format("%s_%s", jobId, stageId),
+        String.format("%s_%s", jobId, senderStageId),
```

Comment:
i dont know if we still want to keep the senderStage in the jobId format. since now you have the sender/receiver id already in the mailboxIdentifier impl

Reply:
Yeah I wanted to change but it would cause a downtime during a rollout. Can we follow up on this in a separate PR?

Reply:
it will be backward incompatible anyway since you changed the JSON format? or are we doing unknown field as ignore?

Reply:
Yeah the json ignore thing should help with no downtimes during rollout.